tokio-reactor/src/lib.rs

```diff
@@ -160,54 +160,41 @@ fn _assert_kinds() {
 
 // ===== impl Reactor =====
 
-/// Set the default reactor for the duration of the closure
-///
-/// # Panics
-///
-/// This function panics if there already is a default reactor set.
-pub fn with_default<F, R>(handle: &Handle, f: F) -> R
-where
-    F: FnOnce() -> R,
-{
-    // Ensure that the executor is removed from the thread-local context
-    // when leaving the scope. This handles cases that involve panicking.
-    struct Reset;
-
-    impl Drop for Reset {
-        fn drop(&mut self) {
-            CURRENT_REACTOR.with(|current| {
-                let mut current = current.borrow_mut();
-                *current = None;
-            });
-        }
-    }
-
-    // This ensures the value for the current reactor gets reset even if there
-    // is a panic.
-    let _r = Reset;
+#[derive(Debug)]
+///Guard that resets current reactor on drop.
+pub struct CurrentReactorReset {}
```

Comment:
What if we name this `CurrentGuard`? Including `Reactor` in there feels like a bit of a stutter. The `Guard` suffix is taken from `MutexGuard`.

Reply:
Sure, I'll make it shorter